types/tls.go

```diff
@@ -183,3 +166,26 @@ var (
 		TLS_RSA_WITH_AES_256_CBC_SHA:                  "AES256-SHA",
 	}
 )
+
+func (c *TLSCipherSuite) String() string {
+	return fmt.Sprintf("%s", *c)
+}
+
+// TODO: Should this return an errgroup instead of only the first error?
+func ValidateEnvoyCipherSuites(cipherSuites []TLSCipherSuite) error {
+	for _, c := range cipherSuites {
+		if _, ok := envoyTLSCipherSuiteStrings[c]; !ok {
+			return fmt.Errorf("no matching Envoy TLS cipher suite found for %s", c.String())
+		}
+	}
+
+	return nil
+}
+
+func MarshalEnvoyTLSCipherSuiteStrings(cipherSuites []TLSCipherSuite) []string {
+	cipherSuiteStrings := []string{}
+	for _, c := range cipherSuites {
+		cipherSuiteStrings = append(cipherSuiteStrings, envoyTLSCipherSuiteStrings[c])
+	}
+	return cipherSuiteStrings
+}
```

Comment:
I think these exported functions tiptoe the line of business logic being in what should be a simple types package. Could they live closer to the packages where they're used?

Reply:
Yea that could makes sense, most appropriate spot would be alongside `envoyTLSVersions` in ` agent/xds/listeners_ingress.go`

Reply:
Can the envoy map live in `agent/xds` or is there reason to keep it in `types`? It feels weird to have envoy specific types/functions live in this package.

Reply:
I think moving `envoyTLSCipherSuiteStrings` and any functions that reference envoy into `agent/xds` makes sense.

Reply:
Was thinking about this a little more from the perspective of Consul API Gateway, and I'd like to avoid re-implementing/maintaining this list in multiple projects.

Ideally we could move this to a common xDS library - I had been thinking we could expose the validation function through `consul/api` in the meantime, but I guess that can't reference anything in this file anyway 😞

Reply:
Welp, started trying to do this and hit the issue that `MarshalEnvoyTLSCipherSuiteStrings` is called only in `agent/xds/listener_ingress.go` and `ValidateEnvoyCipherSuites` is called only in `agent/structs/config_entry_gateways.go`, but both reference the `envoyTLSCipherSuiteStrings` lookup map which makes this a little more difficult.